### PR TITLE
operator: Convert logrus to slog

### PR DESCRIPTION
**Pull request by sayboras**

### CODE REVIEW

File: operator/api/server.go
Comment (bimmlerd) [inline operator/api/server.go]: Did this ever do anything?  the `With` family of methods returns a new logger with added context, which is discarded in this case (both in new and old code)

**Review by bimmlerd — APPROVED** [review]
contributing change lgtm; minor nit when glancing through code

File: operator/api/server.go
Comment (sayboras) [reply operator/api/server.go]: good spot, will remove this one.

**Review by doniacld — APPROVED** [review]

File: operator/endpointgc/gc.go
Comment (doniacld) [inline operator/endpointgc/gc.go:157]: nit: This switch of order's parameters is not really belonging to this commit.

**Review by thorn3r — APPROVED** [review]

File: operator/endpointgc/gc.go
Comment (sayboras) [reply operator/endpointgc/gc.go:157]: yeah, I was reading the code and find it un-easy with context param last 😅

**Review by dylandreimerink — APPROVED** [review]
Seems good for my CODEOWNERS

**Review by harsimran-pabla — APPROVED** [review]

**Review by pippolo84 — APPROVED** [review]
Nice, thanks! :rocket:

File: operator/watchers/logfields.go
Comment (giorio94) [inline operator/watchers/logfields.go:11]: @sayboras This appears not to correctly respect the settings (e.g., debug log), because the default logger is completely replaced (rather than being mutated) during the configuration phase: https://github.com/cilium/cilium/blob/91a844c7eb3971d2e0130dcb390ffe733cc71fa8/pkg/logging/slog.go#L77-L88

Could you please take a look?

Comment (sayboras) [reply operator/watchers/logfields.go:11]: Ah thanks, let me double check

Comment (bimmlerd) [reply operator/watchers/logfields.go:11]: Argh, yeah, I committed that crime originally; I should have caught this. IIRC the idea was that you're supposed to use the slog.Logger provided by hive, which would then be a stable reference

(the problem is that slog doesn't allow you to mutate the logger)

Comment (giorio94) [reply operator/watchers/logfields.go:11]: Yeah, I guess the tricky part is for the modules that have not yet been converted to hive yet. I wonder whether it could make sense to maybe unexport that variable (and provide a safer getter if necessary) to prevent potentially storing a reference to the uninitialized version.

**Pull request merged**
